upgradability/forward-call/address_book/lib.rs

```diff
@@ -0,0 +1,71 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod address_book {
+    use ink_storage::{traits::SpreadAllocate, Mapping};
+    use ink_prelude::string::String;
+
+    const MAX_INFO_SIZE: usize = 20;
+
+    #[ink(storage)]
+    #[derive(SpreadAllocate)]
+    pub struct AddressBook {
+        contact_info: Mapping<AccountId, String>,
+    }
+
+    impl AddressBook {
+        /// Initializes an empty contact info DB.
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            ink_lang::utils::initialize_contract(|_| {})
+        }
+
+        /// Sets contact info of the caller.
+        /// Returns true if this operation was successful.
+        #[ink(message)]
+        pub fn set_info(&mut self, info: String) -> bool {
+            if info.len() > MAX_INFO_SIZE {
+                return false;
```

Comment:
if this can fail it's more idiomatic to make the tx return a Result (and an Err variant in here)